bodyswap.lua

```diff
@@ -122,8 +135,88 @@ function swapAdvUnit(newUnit)
     df.global.adventure.player_id = newNem.id
     df.global.world.units.adv_unit = newUnit
     oldUnit.idle_area:assign(oldUnit.pos)
+    local pos = xyz2pos(dfhack.units.getPosition(newUnit))
+
+    -- reveal the tiles around the bodyswapped unit
+    reveal_tile(xyz2pos(pos.x-1, pos.y-1, pos.z))
+    reveal_tile(xyz2pos(pos.x,   pos.y-1, pos.z))
+    reveal_tile(xyz2pos(pos.x+1, pos.y-1, pos.z))
+    reveal_tile(xyz2pos(pos.x-1, pos.y,   pos.z))
+    reveal_tile(pos)
+    reveal_tile(xyz2pos(pos.x+1, pos.y,   pos.z))
+    reveal_tile(xyz2pos(pos.x-1, pos.y+1, pos.z))
+    reveal_tile(xyz2pos(pos.x,   pos.y+1, pos.z))
+    reveal_tile(xyz2pos(pos.x+1, pos.y+1, pos.z))
+
+    dfhack.gui.revealInDwarfmodeMap(pos, true)
+end
 
-    dfhack.gui.revealInDwarfmodeMap(xyz2pos(dfhack.units.getPosition(newUnit)), true)
+-- shamelessly copy pasted from gui/sitemap.lua
+local function get_unit_choices()
+    local choices = {}
+    for _, unit in ipairs(df.global.world.units.active) do
+        if not dfhack.units.isActive(unit) or
+            dfhack.units.isHidden(unit)
+        then
+            goto continue
```

Comment:
do you also need to exclude other units in your party?

Reply:
Being able to swap to your party members is an intended use case